Libplanet.Net/Swarm.cs

```diff
@@ -97,6 +98,18 @@ public partial class Swarm<T> : IDisposable
             Transport.ProcessMessageHandler.Register(ProcessMessageHandlerAsync);
             PeerDiscovery = new KademliaProtocol(RoutingTable, Transport, Address);
 
+            // Regulate heavy tasks. Treat negative value as 0.
+            var resourceRegulationOptions = Options.ResourceRegulationOptions;
+            int limit = resourceRegulationOptions.MaxTransferBlocksTaskCount;
+            _transferBlocksSemaphore = limit > 0
+                ? new Semaphore(limit, limit)
+                : null;
```

Comment:
Introducing a new static helper method to convert int to semaphore may help for readability.